lib/ash/resource/validation/compare.ex

```diff
@@ -8,25 +8,40 @@ defmodule Ash.Resource.Validation.Compare do
 
   @opt_schema [
     greater_than: [
-      type: {:or, [:integer, :atom, :float, {:struct, Decimal}, {:fun, 0}]},
+      type: :any,
```

Comment:
@zachdaniel I wasn't sure how to define the type correctly for this now that we want to support `{:value, number}` for this. if you have suggestions on how to update this, happy to add it

Reply:
Lets do something like this (for docs) `{:or, [:any, {:tagged_tuple, :value, :any}, {:tagged_tuple, :ref, :atom}]}`